src/input-number/src/InputNumber.tsx

```diff
@@ -362,44 +366,47 @@ export default defineComponent({
           onMousedown={this.handleMouseDown}
         >
           {{
-            suffix: () => [
-              <NButton
-                text
-                disabled={!this.minusable || this.disabled}
-                focusable={false}
-                builtinThemeOverrides={this.buttonThemeOverrides}
-                onClick={this.handleMinusClick}
-                ref="minusButtonInstRef"
-              >
-                {{
-                  default: () => (
-                    <NBaseIcon clsPrefix={mergedClsPrefix}>
-                      {{
-                        default: () => <RemoveIcon />
-                      }}
-                    </NBaseIcon>
-                  )
-                }}
-              </NButton>,
-              <NButton
-                text
-                disabled={!this.addable || this.disabled}
-                focusable={false}
-                builtinThemeOverrides={this.buttonThemeOverrides}
-                onClick={this.handleAddClick}
-                ref="addButtonInstRef"
-              >
-                {{
-                  default: () => (
-                    <NBaseIcon clsPrefix={mergedClsPrefix}>
-                      {{
-                        default: () => <AddIcon />
-                      }}
-                    </NBaseIcon>
-                  )
-                }}
-              </NButton>
-            ]
+            suffix: () =>
+              this.showButton
```

Comment:
extract `this.showButton` outside.

this.showButton ? {
  suffix: ...  
} : undefined